osu.Game/Screens/Play/KeyCounter.cs

```diff
@@ -69,82 +67,36 @@ public void Decrement()
             CountPresses--;
         }
 
-        //further: change default values here and in KeyCounterCollection if needed, instead of passing them in every constructor
-        public Color4 KeyDownTextColor { get; set; } = Color4.DarkGray;
-        public Color4 KeyUpTextColor { get; set; } = Color4.White;
-        public double FadeTime { get; set; }
-
-        protected KeyCounter(string name)
+        public abstract partial class InputTrigger : Component
         {
-            Name = name;
-        }
+            private KeyCounter? target;
 
-        [BackgroundDependencyLoader(true)]
-        private void load(TextureStore textures)
-        {
-            Children = new Drawable[]
+            public KeyCounter Target
```

Comment:
The circularity of this is giving me a headache. `KeyCounter`s have instances of `InputTrigger`, which are exposed so that the `RulesetInputManager` can access the `InputTrigger`s and call methods on them, which... end up performing actions on the `KeyCounter` again?

I'd like to see looser coupling here. Maybe have `InputTrigger` only expose events and be blissfully unaware of the key counter, and the counter can subscribe to the events and do what it needs to do.